apps/passport-client/src/broadcastChannel.ts

```diff
@@ -0,0 +1,46 @@
+import { Action } from "./dispatch";
+
+const CHANNEL_NAME = "zupass_broadcast_channel";
+// The event message that prompts other tabs to refresh their local state
+const UPDATE_STATE_MESSAGE = "update_state_on_other_tabs";
+let channel: BroadcastChannel | null = null;
+
+/**
+ * Force other tabs in the current browser session to load their AppState from
+ * the saved localStorage values. This ensures that when we perform actions that
+ * change localStorage like changing password, this update is also reflected in
+ * the in-browser memory of other tabs within the session.
```

Comment:
I would specify that this doesn't force them to load the *entire* app state, just the parts that are relevant. reading this comment scared me a little bit. eg. another thing that lives in app state is which modal is open - idk if that's something that should be shared between tabs, though it would be kind of cool.